File: Ejercicios/5. Duck/Net/After/Duck.Tests/DecoyDuckTests.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Duck.Tests
{
    [TestClass]
    public class DecoyDuckTests
    {
        private Duck duck;

        [TestInitialize]
        public void Setup()
        {
            duck= new DecoyDuck();
        }

        [TestMethod]
        public void CantFly()
        {
            duck.FlyBehavoir=new FlyNoWay();
            Assert.AreEqual(DuckMessages.CantFly,duck.Fly());
        }

        [TestMethod]
        public void Squeak()
        {
            duck.QuackBehavior=new MuteQuack();
            Assert.AreEqual(DuckMessages.Silence, duck.Quack());
        }


        [TestMethod]
        public void Display()
        {
            Assert.AreEqual(DuckMessages.ModelDisplay, duck.Display());
        }
    }
}

File: Ejercicios/5. Duck/Java/After/src/test/duck/DecoyDuckTests.java
package duck;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

public class DecoyDuckTests {
    private Duck duck;

    @Before
    public void Setup()
    {
        duck= new DecoyDuck();
    }

    @Test
    public void cantFly()
    {
        assertEquals(DuckMessages.CantFly,duck.fly());
    }

    @Test
    public void squeak()
    {
        assertEquals(DuckMessages.Silence, duck.quack());
    }


    @Test
    public void display()
    {
        assertEquals(DuckMessages.ModelDisplay, duck.display());
    }
}
